### PR TITLE
Implemented readme

### PR DESCRIPTION
**Pull request by jordanhudgens**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,29 +1,314 @@
...
-migrations
-models
-controllers
-resources
-cover the magic route resources but we'll cover that more in depth later.
-choose the smallest generate that provides you with the files you need, don't over generate, the less code in your app the better. never use scaffolds, it's too much code, learn the pattern and generate a resource.
+## Abusing Generators
 
-Don't cover scaffolds in this...
+So if generators are so amazing, why wouldn't we always want to use them all of the time? I'll answer your question with another question: a chainsaw is a wonderful tool, why wouldn't you use it for every job you have around the house? The answer to both questions is the same: great tools are only great tools if they are matched with the right task. In the same way that you should only use a chainsaw when you have a job that requires it, generators should only be used when they are needed.
```
Comment (jmburges) [inline README.md]: :+1:

File: README.md [hunk 2]
```diff
@@ -1,29 +1,314 @@
...
+
+Extending our example from above, why wouldn't you use a chainsaw to build a model airplane?
+
+![Chainsaw Example](https://s3.amazonaws.com/flatiron-bucket/readme-lessons/chainsaw_example.jpg)
```
Comment (jmburges) [inline README.md]: hahahahaha amazing

File: README.md [hunk 3]
```diff
@@ -1,29 +1,314 @@
...
+
+In the same way as our chainsaw example, certain generators create quite a bit of code and if that code is not going to be used it will clutter the application code and cause confusion for future developers. A few years ago I took over as the lead developer for a large legacy Rails application. The previous developer had relied on generators, even when they shouldn't have been used, and the end result was that it took months to simply figure out what code was being used and what was 'garbage' code that simply came from the generators.
+
+So when is the right time to use a generator? After we've gone over the actions of each of the generators I think the answer to this query will become readily apparent and we'll walk through some case studies to help understand when each type of generator is beneficial.
+
+
+## Rails Generate
+
+Each of the generators will follow the syntax:
```
Comment (jmburges) [inline README.md]: explain you are typing these in terminal

File: README.md [hunk 4]
```diff
@@ -1,29 +1,314 @@
...
+* Models
+* Controllers
+* Resources
+
+
+## Migration Generators
+
+Up until this point we have been creating our migrations by hand, this has been beneficial since it's important to understand how migrations work, however Rails has a great set of migration generators with conventions that can help make managing the database schema very efficient.
+
+Let's start using database migrations in our case study application and update the `posts` table. To add a new column we can use the following command:
```
Comment (jmburges) [inline README.md]: include the application "to date" in with this repo

Comment (jmburges) [inline README.md]: also explain the new column you want to create so the following generate command makes sense

File: README.md [hunk 5]
```diff
@@ -1,29 +1,314 @@
...
+```
+
+We can simply add in the `change_column` method like this: `change_column :posts, :post_status, :string` and after running `rake db:migrate` our schema will be updated.
+
+[Full migration documentation](http://api.rubyonrails.org/classes/ActiveRecord/Migration.html)
+
+
+# Model Generators
+
+This is a generator type that I use regularly, it does a great job of creating the core code needed to create a model and associated database table, without adding a lot of bloat to the application. Let's add a new model to the app called `Author`, we can use the model generator with the following CLI command:
```
Comment (jmburges) [inline README.md]: also here tell them that this authro should have a name, genre, etc

File: README.md [hunk 6]
```diff
@@ -1,29 +1,314 @@
...
+Author.create!(name: "Stephen King", genre: "Horror", bio: "Bio details go here")
+=> #<Author id: 1, name: "Stephen King", genre: "Horror", bio: "Bio details go here", created_at: "2015-11-27 22:59:14", updated_at: "2015-11-27 22:59:14"> 
+```
+
+So it looks like our model has been created properly, and as you can see, this particular generator created a a few different pieces of functionality with a single command and it did it with creating minimal code bloat.
+
+
+## Controller Generators
+
+Controller generators are great if you are creating static views or non crud related features (we'll walk through why this is the case shortly). Let's create an `admin` controller that will manage the data flow and view rendering for our admin dashboard pages:
```
Comment (jmburges) [inline README.md]: cap CRUD

Comment (jmburges) [inline README.md]: also why are they great for non-crud/static?

**Comment by jmburges** [comment]
back at you

**Comment by jordanhudgens** [comment]
@jmburges This one has been revised and the app added.

**Comment by jmburges** [comment]
Hey @jordanhudgens so close! So the tests all pass when you clone! There should be non passing tests to make sure they actually create the migrations you ask them to make. If you can do that, then we'll be good to go and you can merge away!

**Pull request merged**